core/textile/collections.go

```diff
@@ -230,3 +248,84 @@ func (tc *textileClient) initBucketCollection(ctx context.Context) (context.Cont
 
 	return metaCtx, dbID, nil
 }
+
+// initializes bucket path encryption collection inside the thread provided.
+func (tc *textileClient) initBucketPathEncryptionCollection(ctx context.Context, dbId thread.ID) error {
+	if err := tc.threads.NewCollection(ctx, dbId, db.CollectionConfig{
+		Name:   pathEncryptionCollectionName,
+		Schema: util.SchemaFromInstance(&BucketPathEncryptionSchema{}, false),
+		Indexes: []db.Index{
+			{
+				Path:   "bucket_slug",
+				Unique: false,
+			},
+		},
+	}); err != nil {
+		log.Debug("Failed to create bucket encryption collection", "err:"+err.Error(), "dbId:"+string(dbId))
+		return err
+	}
+
+	return nil
+}
+
+func (tc *textileClient) findBucketPathEncryption(
+	ctx context.Context,
+	bucketSlug, path string,
+) (*BucketPathEncryptionSchema, error) {
+	metaCtx, metaDbID, err := tc.initBucketCollection(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := tc.threads.Find(
+		metaCtx,
+		*metaDbID,
+		pathEncryptionCollectionName,
+		db.Where("bucket_slug").Eq(bucketSlug).And("_id").Eq(path),
+		&BucketPathEncryptionSchema{},
+	)
+
+	if err != nil {
+		log.Debug("Failed to find bucket path encryption", "error:"+err.Error(), "bucket:"+bucketSlug, "path:"+path)
```

Comment:
Maybe you can use log.Error

Reply:
Yeah, I don't know why I used debug here 😅